pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,291 @@
+                0.        +0.j        , 0.        +0.j        ],
+               [0.        +0.j        , 1.81677345-0.57695852j,
+                0.        +0.j        , 0.        +0.j        ],
+               [0.        +0.j        , 0.        +0.j        ,
+                1.81677345+0.57695852j, 0.        +0.j        ],
+               [0.        +0.j        , 0.        +0.j        ,
+                0.        +0.j        , 1.81677345+0.57695852j]])
+    """
+
+    _eigs = {}  # cache eigen vectors and values like in qml.Hermitian
+
+    def __init__(self, *summands, do_queue=True, id=None):
+        """Initialize a Symbolic Operator class corresponding to the Sum of operations."""
+        self._name = "Sum"
+        self._id = id
+        self.queue_idx = None
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
+        self.data = [s.parameters for s in summands]
+        self._wires = qml.wires.Wires.all_wires([s.wires for s in summands])
+
+        if do_queue:
+            self.queue()
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+        copied_op.summands = tuple(s.__copy__() for s in self.summands)
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "summands"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    @property
+    def batch_size(self):
+        """Batch size of input parameters."""
+        raise ValueError("Batch size is not defined for Sum operators.")
+
+    @property
+    def ndim_params(self):
+        """ndim_params of input parameters."""
+        raise ValueError("Dimension of parameters is not currently implemented for Sum operators.")
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def num_params(self):
+        return sum(op.num_params for op in self.summands)
+
+    @property
+    def is_hermitian(self):
+        """If all of the terms in the sum are hermitian, then the Sum is hermitian."""
+        return all(s.is_hermitian for s in self.summands)
+
+    def terms(self):
+        r"""Representation of the operator as a linear combination of other operators.
+
+        .. math:: O = \sum_i c_i O_i
+
+        A ``TermsUndefinedError`` is raised if no representation by terms is defined.
+
+        .. seealso:: :meth:`~.Operator.compute_terms`
+
+        Returns:
+            tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
+            and list of operations :math:`O_i`
+        """
+        return [1.0] * len(self.summands), self.summands
```

Comment:
Looks like this might be missing coverage.

Reply:
Ahh great catch!